specutils/manipulation/resample.py

```diff
@@ -0,0 +1,409 @@
+        # match the input spectrum it won't work right, will have to think
+        # more about how to handle that... could convert before and after
+        # calculation, which is probably easiest. Matrix math algorithm is
+        # geometry based, so won't work to just let quantity math handle it.
+        resampled_spectrum = Spectrum1D(flux=out_flux,
+                                        spectral_axis=np.array(fin_lamb) * orig_spectrum.spectral_axis_unit,
+                                        uncertainty=out_uncertainty)
+
+        return resampled_spectrum
+
+
+class LinearInterpolatedResample(ResampleBase):
+    """
+    This resample algorithim uses a linear interpolator.
+
+    Examples
+    --------
+
+    To resample an input spectrum to a user specified dispersion grid using
+    linear interpolation:
+
+    >>> import numpy as np
+    >>> import astropy.units as u
+    >>> from specutils import Spectrum1D
+    >>> from specutils.manipulation import LinearInterpolatedResample
+    >>> input_spectra = Spectrum1D(
+    ...     flux=np.array([1, 3, 7, 6, 20]) * u.mJy,
+    ...     spectral_axis=np.array([2, 4, 12, 16, 20]) * u.nm)
+    >>> resample_grid = np.array([1, 5, 9, 13, 14, 17, 21, 22, 23])
+    >>> fluxc_resample = LinearInterpolatedResample()
+    >>> output_spectrum1D = fluxc_resample(input_spectra, resample_grid) # doctest: +IGNORE_OUTPUT
+    """
+    def __init__(self, bin_edges='nan_fill'):
+        super().__init__(bin_edges)
+
+    def __call__(self, orig_spectrum, fin_lamb):
+        """
+        Return the resulting `~specutils.Spectrum1D` of the resampling.
+        """
+        return self.resample1d(orig_spectrum, fin_lamb)
+
+    def _interpolation(self, orig_dispersion, flux, fin_lamb):
+        """
+        Use specified interpolation to calculated resampled
+        flux.
+
+        Parameters
+        ----------
+        orig_dispersion : ndarray
+            The original dispersion array.
+        flux: ndarray
+            The flux array from the input Spectrum1D
+        fin_lamb : ndarray
+            The desired dispersion array.
+
+        Returns
+        -------
+        resample_flux : ndarray
+            The resampled flux array generated from the interpolation.
+
+        """
+        return np.interp(fin_lamb, orig_dispersion, flux, left=np.nan, right=np.nan)
+
+    def resample1d(self, orig_spectrum, fin_lamb):
+        """
+        Call interpolation, repackage new spectra
+
+
+        Parameters
+        ----------
+        orig_spectrum : `~specutils.Spectrum1D`
+            The original 1D spectrum.
+        fin_lamb : ndarray
+            The desired dispersion array.
+
+        Returns
+        -------
+        resample_spectrum : `~specutils.Spectrum1D`
+            An output spectrum containing the resampled `~specutils.Spectrum1D`
+        """
+        out_flux = self._interpolation(orig_spectrum.spectral_axis, orig_spectrum.flux,
```

Comment:
I agree it's out-of-scope to implementation uncertainties right now, but perhaps you should add a warning if the input spectrum has an uncertainty?  I.e.:
```suggestion
        if orig_spectrum.uncertainty is not None:
            warn("Linear interpolation currently does not propogate uncertainties")
        out_flux = self._interpolation(orig_spectrum.spectral_axis, orig_spectrum.flux,
```